plugin/sampling/strategystore/factory.go

```diff
@@ -23,30 +23,32 @@ import (
 	"go.uber.org/zap"
 
 	"github.com/jaegertracing/jaeger/cmd/collector/app/sampling/strategystore"
+	"github.com/jaegertracing/jaeger/pkg/distributedlock"
 	"github.com/jaegertracing/jaeger/plugin"
+	"github.com/jaegertracing/jaeger/plugin/sampling/strategystore/adaptive"
 	"github.com/jaegertracing/jaeger/plugin/sampling/strategystore/static"
+	"github.com/jaegertracing/jaeger/storage/samplingstore"
 )
 
-const (
-	staticStrategyStoreType = "static"
-)
+// Kind is a datatype holding the type of strategy store.
+type Kind string
 
-var allSamplingTypes = []string{staticStrategyStoreType} // TODO support adaptive
+var allSamplingTypes = []Kind{"static", "adaptive"}
```

Comment:
maybe define these as constants of Kind type? The literals are used in more than one place.